### ISSUE TITLE
Synchronize multiple devices time signature through Ableton Link

### ISSUE THREAD

**Issue by Carvahalll** [Ableton/link#144]
Hello guys, I'm writing a metronome app for android. The idea is to sync multiple android devices and Ableton instances to share the same tempo AND time signature.
The idea is to allow all musicians of a band to use their phone as a metronome app during practice (or even a concert) while having Ableton propagate the current BPM, phase and time signature for each song in the set.
The Ableton link protocol works amazing for syncing BPM and start/stop, but I'm not able to find a way to sync time signature without using an additional protocol like OSC (which would be cumbersome as Ableton doesn't support OSC natively anyways). I get, that the idea of the link protocol is for each device  to have their own quantum, but do I understand correctly that there is intentionally no function that allows each device to broadcasts their quantum? This would allow the other devices to decide if they want to adapt it or not.
Or am I missing something?

**Comment by cdi-ableton** [comment]
Yes, the quantum is not shared or maintained by Link. It's only used by Link's API to map beat time to host time, allowing Link peers to run in different meters. This allows a polymeter sequencer to query beat times for multiple quanta without affecting other peers. 
It is conceivable to optionally synchronize the quantum, such as start/stop events, and we'll add it to the Link feature wishlist. But for now, this work is not planned.

You can send Ableton Live's time signature via OSC using Max-for-Live.

**Comment by WhizKid21** [comment]
Just adding another real-world use case in support of this request.

I’m a solo live performer using **OnSong on an iPad together with the Soundbrenner app and a Soundbrenner Core 2 wearable metronome**.

My intended workflow is:

**OnSong chart → Ableton Link → Soundbrenner → Core 2**

Each song in OnSong already contains the correct **tempo and time signature**. When I select a song, Ableton Link does a great job of allowing the tempo to propagate automatically to Soundbrenner. What is missing is the meter.

For example, I may move through a live set containing songs in:

* 4/4
* 3/4
* 6/8
* 12/8

OnSong already knows that information. Soundbrenner is also capable of operating in those time signatures, and the meter is important because it determines the downbeat/accent pattern I feel through the Core 2.

Ideally, simply selecting the next song in OnSong would allow **both BPM and time signature to follow automatically**.

Without that capability, I either have to manually change the meter in Soundbrenner or maintain a second parallel Soundbrenner setlist and advance both applications separately. In a live performance environment, that is exactly the sort of additional interaction I am trying to eliminate.

I completely understand the Link philosophy of allowing individual participants to use their own quantum and musical structure. I would not want shared meter to override that flexibility.

What would be extremely useful is an **optional shared time-signature/meter value** — for example numerator and denominator — that a Link participant could publish, and that receiving applications could choose either to follow or ignore.

That would preserve Link’s existing flexibility while allowing applications such as OnSong, DAWs, metronomes, lighting systems, backing-track applications, etc. to share musical meter when it is useful.

In my particular case, both endpoints already understand the information:

**OnSong knows the song is 6/8 → Soundbrenner knows how to operate in 6/8.**

Link is simply unable to communicate that piece of information between them.

I’d love to see optional time-signature metadata added to Link in a future version. It would make Link substantially more useful for live musicians whose sets involve multiple meters, not just tempo changes.

Thanks for considering it — and thanks for Link in general. The tempo synchronization portion of this workflow already works extremely well.
